### DISCUSSION TITLE
Adding extension methods on C# object: is it a code smell or not?

### DISCUSSION THREAD

**Discussion by metaturso** [dotnet/roslyn#67672] [Q&A] [answered]
In Framework Design Guidelines, Section 5.6, on p165, as well as [this article](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/extension-methods) on Microsoft's website, the authors advise against defining extension methods on `System.Object`, because they wouldn't work in late-bound languages such as VB.

Sonarsource's [RSPEC-4225](https://rules.sonarsource.com/csharp/RSPEC-4225) analyzer extends this interpretation to C# also. From the wording on the Microsoft article alone, I can't say that defining extension methods on a C# `object` is undesirable.

Leaving aside the different types of binding in the two languages, I would like to understand the opinion of the Roslyn users and its team about the practice of defining extension methods on `object` in C#. Is it undesirable or not?

**Comment by CyrusNajmabadi** [comment, marked as answer]
It's up to you if it's undesirable.  The language supports it, but leaves it to the individual or team to decide if that's ok with them or not.
